www-client/elinks/elinks-0.16.1.1-r3.ebuild

```diff
@@ -0,0 +1,187 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..11} )
+LUA_COMPAT=( lua5-{1,2,3,4} luajit )
+
+inherit flag-o-matic meson lua-single python-single-r1
+
+DESCRIPTION="Advanced and well-established text-mode web browser"
+HOMEPAGE="http://elinks.or.cz/"
+
+if [[ ${PV} == *9999 ]] ; then
+	EGIT_REPO_URI="https://github.com/rkd77/felinks"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/rkd77/elinks/releases/download/v${PV}/${P}.tar.xz"
+
+	KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x64-solaris"
+fi
+
+LICENSE="GPL-2"
+SLOT="0"
+IUSE="bittorrent brotli bzip2 debug finger ftp gopher gpm gnutls guile idn"
+IUSE+=" javascript lua lzma +mouse nls nntp perl python samba ssl test tre unicode X xml zlib zstd"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="
+	lua? ( ${LUA_REQUIRED_USE} )
+	python? ( ${PYTHON_REQUIRED_USE} )
+"
+
+RDEPEND="
+	>=sys-libs/ncurses-5.2:=[unicode(+)]
+	brotli? ( app-arch/brotli:= )
+	bzip2? ( >=app-arch/bzip2-1.0.2 )
+	gpm? (
+		>=sys-libs/gpm-1.20.0-r5
+	)
+	guile? ( >=dev-scheme/guile-1.6.4-r1[deprecated] )
+	idn? ( net-dns/libidn:= )
+	javascript? (
+		dev-cpp/libxmlpp:5.0
+		dev-lang/mujs:=
+	)
+	lua? ( ${LUA_DEPS} )
+	lzma? ( app-arch/xz-utils )
+	perl? ( dev-lang/perl:= )
+	python? ( ${PYTHON_DEPS} )
+	samba? ( net-fs/samba )
+	ssl? (
+		!gnutls? ( dev-libs/openssl:= )
+		gnutls? ( net-libs/gnutls:= )
+	)
+	tre? ( dev-libs/tre )
+	X? (
+		x11-libs/libX11
+		x11-libs/libXt
+	)
+	xml? ( >=dev-libs/expat-1.95.4 )
+	zlib? ( >=sys-libs/zlib-1.1.4 )
+	zstd? ( app-arch/zstd:= )
+"
+DEPEND="${RDEPEND}
+	X? ( x11-base/xorg-proto )"
+BDEPEND="
+	virtual/pkgconfig
+	nls? ( sys-devel/gettext )
+	test? (
+		net-dns/libidn
+	)
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-0.16.1.1-perl-5.38.patch
+)
+
+pkg_setup() {
+	use lua && lua-single_pkg_setup
+
+	use python && python-single-r1_pkg_setup
+}
+
+src_configure() {
+	# This file is severely broken w.r.t. strict-aliasing and upstream acknowledges it:
+	# https://github.com/rkd77/elinks/blob/d05ce90b35d82109aab320b490e3ca54aa6df057/src/util/lists.h#L14
+	# https://github.com/rkd77/elinks/blob/d05ce90b35d82109aab320b490e3ca54aa6df057/src/meson.build#L44
+	#
+	# Although they force fno-strict-aliasing, they do so inconsistently and not for the testsuite (!!!).
+	# Just add it again.
+	#
+	# DO not trust the LTO either.
+	append-flags -fno-strict-aliasing
+	filter-lto
+
+	local emesonargs=(
+		-Ddocdir="${EPREFIX}"/usr/share/doc/${PF}
+		-Dhtmldoc=false
+		-Dpdfdoc=false
+		-Dapidoc=false
+		-D88-colors=true
+		-D256-colors=true
+		$(meson_use bittorrent)
+		$(meson_use brotli)
+		$(meson_use bzip2 bzlib)
+		$(usex debug '-Ddebug=true' '-Dfastmem=true')
+		$(meson_use finger)
+		$(meson_use ftp)
```

Comment:
This also needed `$(meson_use ftp libcurl)` otherwise FTP feature won't build. I'm just not sure if it's still necessary now now that upstream has already fixed a few code related to FTP.